app/routes.test.js

```diff
@@ -0,0 +1,402 @@
+import request from 'supertest';
+import { App } from '../app';
+import routes from './routes';
+import * as dashboardControllers from './pages/dashboard/controller';
+import * as sectionControllers from './pages/section/controller';
+import * as subsectionControllers from './pages/dashboard/subDashboards/controller';
+import * as previewControllers from './pages/preview/controller';
+
+jest.mock('./logger');
+
+const mockDashboardContext = {
+  title: 'Dashboard title',
+  mainAdvice: undefined,
+  additionalAdvice: undefined,
+  previewUrl: '/',
+  submitForModerationUrl: '/',
+  returnToDashboardUrl: '/',
+  errors: undefined,
+  sectionGroups: [
+    {
+      id: 'sectiongroup-id',
+      title: 'Sectiongroup Title',
+      sections: [],
+    },
+  ],
+};
+const mockSectionContext = {
+  title: 'Section Title',
+  submitActionUrl: '/s',
+  mainAdvice: 'Section Main Advice',
+  additionalAdvice: [
+    'Section additional advice',
+  ],
+  warningAdvice: undefined,
+  returnToDashboardUrl: '/',
+  submitText: 'submitText',
+  errors: undefined,
+  questions: [
+    {
+      id: 'a-question',
+      type: 'bulletpoint-list',
+      mainAdvice: undefined,
+      additionalAdvice: undefined,
+      footerAdvice: undefined,
+      fields: [],
+    },
+  ],
+};
+const mockSectionPostData = {
+  'a-question': [
+    'bullet 1',
+    '',
+    '',
+    '',
+    '',
+    '',
+    '',
+    '',
+    '',
+    '',
+  ],
+};
+const mockPostSectionContext = {
+  title: 'Post Section Title',
+  submitActionUrl: '/',
+  mainAdvice: 'post section main advice',
+  additionalAdvice: [
+    'post section additional advice',
+  ],
+  warningAdvice: undefined,
+  returnToDashboardUrl: '/',
+  submitText: 'submitText',
+  errors: [
+    {
+      text: 'This is over the character limit',
+      href: '#',
+    },
+  ],
+  questions: [
+    {
+      id: 'a-question',
+      type: 'bulletpoint-list',
+      mainAdvice: undefined,
+      additionalAdvice: undefined,
+      footerAdvice: undefined,
+      fields: [],
+    },
+  ],
+};
+const mockPreviewContext = {
+  sections: {
+    'solution-description': { answers: [Object] },
+    features: { answers: [Object] },
+    'contact-details': { answers: [Object] },
+    capabilities: { answers: [Object] },
+  },
+};
+const mockSectionErrorContext = {
+  ...mockSectionContext,
+  errors: [
+    {
+      text: 'This is over the character limit',
+      href: '#',
+    },
+  ],
+};
+
+describe('GET /healthcheck', () => {
+  it('should return the correct status and text', () => {
+    const app = new App().createApp();
+    app.use('/', routes);
+
+    return request(app)
+      .get('/healthcheck')
+      .expect(200)
+      .then((res) => {
+        expect(res.text).toBe('Marketing pages is Running!!!');
+      });
+  });
+});
+
+describe('GET /solution/:solutionId', () => {
+  it('should return the correct status and text if there is no error', () => {
+    dashboardControllers.getMarketingPageDashboardContext = jest.fn()
```

Comment:
I think we can inline this to;
`dashboardControllers.getMarketingPageDashboardContext.mockImplementation(() => Promise.resolve(mockDashboardContext));`
I've done something similar in the controller unit tests

Reply:
in the controller tests this is because you are mocking the whole class for ApiProvider and ManifestProvider so the class is already a jest mock

Reply:
ie the `getMarketingPageDashboardContext` doesn't have a `mockImplementation` method to call

Reply:
ah cool I get it now. Cool leave as is 👍